File: fsharp/rotational-cipher/RotationalCipher.fs
module RotationalCipher

let shiftBy shiftKey c = 
    if System.Char.IsLetter(c) then
        let zero = if System.Char.IsUpper(c) then 'A' else 'a'
        char <| (int c - int zero + shiftKey) % 26 + int zero 
    else 
        c

let rotate shiftKey text = 
    text 
    |> Seq.map (shiftBy shiftKey) 
    |> Seq.toArray 
    |> System.String
    

File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        char[] chars = text.Select(c => char.IsLetter(c) ? shiftBy(c, shiftKey, char.IsUpper(c)) : c).ToArray();
        return new string(chars);
    }
    public static char shiftBy(char c, int shift, bool upper)
    {
        int zero = upper ? 'A' : 'a';
        return (char)(((c - zero) + shift) % 26 + zero);
    }
}